timeout/timeout.py

```diff
@@ -254,10 +294,20 @@ async def timeout(self, ctx: commands.Context, user: discord.Member, *, reason="
             await ctx.send("I cannot do that due to Discord hierarchy rules.")
             return
 
+        # Create a list containing the timeout role so we can
+        # add the boost role to it if the user is a booster.
+        # This is necessary since you cannot remove the boost
+        # role, so we must ensure we avoid attempting to do so.
+        booster_role = ctx.guild.premium_subscriber_role
+        if user in ctx.guild.premium_subscribers:
+            timeout_roleset = [timeout_role, booster_role]
+        else:
+            timeout_roleset = [timeout_role]
+
         # Check if user already in timeout.
-        # Remove & restore if so, else timeout.
-        if user.roles == [everyone_role, timeout_role]:
+        # Remove & restore if so, else add to timeout.
+        if sorted(user.roles) == sorted([everyone_role] + timeout_roleset):
```

Comment:
```suggestion
        if set(user.roles) == {everyone_role} | {timeout_roleset}:
```
Set comparison seems more straightforward here...